packages/broker/src/helpers/scheduler.ts

```diff
@@ -0,0 +1,20 @@
+export const scheduleAtInterval = async (
+    task: () => Promise<void>, 
+    interval: number, // number of milliseconds to wait after a task is completed
+    executeAtStart: boolean
+): Promise<{ stop: () => void }> => {
+    let timer: NodeJS.Timer|undefined
+    const scheduleNext = () => {
+        timer = setTimeout(async () => {
+            await task()
+            scheduleNext()
```

Comment:
You need to check whether `stop`  was invoked after the above `await` finishes. Otherwise it's possible that the interval continues.

Reply:
Fixed